sdk/batch/azure-batch/azure/batch/custom/custom_errors.py

```diff
@@ -8,12 +8,30 @@ class CreateTasksErrorException(Exception):
     """ Aggregate Exception containing details for any failures from a task add operation.
 
     :param str message: Error message describing exit reason
-    :param [~TaskAddParameter] pending_task_list: List of tasks remaining to be submitted.
+    :param [~TaskAddParameter] pending_tasks: List of tasks remaining to be submitted.
     :param [~TaskAddResult] failure_tasks: List of tasks which failed to add
     :param [~Exception] errors: List of unknown errors forcing early termination
     """
-    def __init__(self, message, pending_task_list=None, failure_tasks=None, errors=None):
-        self.message = message
-        self.pending_tasks = list(pending_task_list)
+    def __init__(self, pending_tasks=None, failure_tasks=None, errors=None):
+        if failure_tasks and errors:
+            self.message =  "Multiple errors encountered. " \
+                            "Check the `failure_tasks` and `errors` properties for additional details"
+        if errors:
+            if len(errors) > 1:
+                self.message = "The SDK encountered an internal error. " \
+                               "Check the `errors` property for the inner exceptions"
+            else:
+                self.message = errors[0].message
+        if failure_tasks:
+            if len(failure_tasks) > 1:
+                self.message = "Multiple client side errors occured when submitting the request. " \
+                               "Check the `failure_tasks` property for details on these tasks."
+            else:
+                result = failure_tasks[0]
+                self.message = "Task with id `%s` failed due to client error - %s::%s" % \
```

Comment:
minor: On this and a few other lines you could break right away and then have more room on each subsequent line for text